File: medium/1288_remove_covered_intervals/solution.cpp
#include <vector>

class Solution {
public:
  int removeCoveredIntervals(std::vector<std::vector<int>> &intervals) {
    sort(intervals.begin(), intervals.end(),
         [](const std::vector<int> &a, const std::vector<int> &b) {
           if (a[0] == b[0])
             return a[1] > b[1];
           return a[0] < b[0];
         });

    int uncoveredIntervalsCount = 0;
    int prevIntervalEnd = -1;

    for (const std::vector<int> &interval : intervals) {
      if (interval[1] > prevIntervalEnd) {
        ++uncoveredIntervalsCount;
        prevIntervalEnd = interval[1];
      }
    }

    return uncoveredIntervalsCount;
  }
};

File: medium/1288_remove_covered_intervals/Solution.java
import java.util.Arrays;

class Solution {
  public int removeCoveredIntervals(int[][] intervals) {
    Arrays.sort(
        intervals,
        (i1, i2) -> {
          final int compare = Integer.compare(i1[0], i2[0]);
          if (compare != 0) {
            return compare;
          }
          return Integer.compare(-i1[1], -i2[1]);
        });

    int uncoveredIntervalsCount = 0;
    int prevIntervalEnd = -1;

    for (final int[] interval : intervals) {
      if (interval[1] > prevIntervalEnd) {
        ++uncoveredIntervalsCount;
        prevIntervalEnd = interval[1];
      }
    }

    return uncoveredIntervalsCount;
  }
}
